File: internship_dec23/jury_solutions/python/b.py
n = int(input())
numbers = input().split()
s = sum([int(x) for x in numbers])

current_sum = 0
for i in range(n - 1):
    current_sum += int(numbers[i])
    if 2 * current_sum == s:
        print('+'.join(numbers[:i + 1]), '=', '+'.join(numbers[i + 1:]), sep='')
        exit()

print(-1)

File: internship_dec23/jury_solutions/cs/b.cs
using System;
using System.Linq;
 
public static void Main()
{
    var n = int.Parse(Console.ReadLine());
    var numbers = Console.ReadLine().Split().Select(int.Parse).ToArray();

    var sumLeft = 0;
    var sumRight = 0;
    var left = 0;

    var right = n - 1;
    while (left <= right)
    {
        if (sumLeft <= sumRight)
        {
            sumLeft += numbers[left];
            left++;
        }
        else
        {
            sumRight += numbers[right];
            right--;
        }
    }

    if (sumRight == sumLeft)
    {
        var leftPart = string.Join("+", numbers.Take(left));
        var rightPart = string.Join("+", numbers.Skip(left));
        Console.WriteLine($"{leftPart}={rightPart}");
    }
    else
        Console.WriteLine(-1);
}